blockchain/state_manager/src/lib.rs

```diff
@@ -50,10 +71,39 @@ where
         Self {
             bs,
             cache: RwLock::new(HashMap::new()),
+            subscriber: None,
+            back_search_wait: None,
+        }
+    }
+
+    //Creates a constructor that passes in a HeadChange subscriber
```

Comment:
Also, can you change all these to doc comments (`///`)? And very specific nit but even when doing regular comments if you could please add a space after the `//` for consistency?